File: Leetcode/palindrome/palindrome_number.c
#include <stdio.h>
int main() {
    int n, reversedN = 0, remainder, originalN;
    //Taking user input
    printf("Enter an integer: ");
    scanf("%d", &n);
    originalN = n;

    // reversed integer is stored in reversedN
    while (n != 0) {
        remainder = n % 10;                        //will give last digit
        reversedN = reversedN * 10 + remainder;     //forming reversed number
        n /= 10;
    }

    // palindrome if orignalN and reversedN are equal
    if (originalN == reversedN)
        printf("%d is a palindrome.", originalN);
    else
        printf("%d is not a palindrome.", originalN);

    return 0;
}

File: Leetcode/Palindrome/Palindrome_Number.java
/*
Java Program to check whether a number is Palindrome or not
An integer is a palindrome if the reverse of that number is equal to the original number.
Example of Palindromes are 11,44,101,121.
*/

public class Solution {
  public boolean isPalindrome(int x) {
    if (x < 0 || x != 0 && x % 10 == 0) {
      return false;
    }

    // reverse process
    int reversed = 0;
    int original = x;
    while (x != 0) {
      reversed = reversed * 10 + x % 10;
      x = x / 10;
    }

    return reversed == original;
  }
}

/*

Example 1:
Input: x = 121
Output: true

Example 2:
Input: x = -121
Output: false

*/
